src/Stack/Upload.hs

```diff
@@ -83,14 +112,15 @@ loadCreds config = do
       writeFilePrivate fp $ lazyByteString lbs
 
       unless (configSaveHackageCreds config) $ do
-        putStrLn "WARNING: You've set save-hackage-creds to false"
-        putStrLn "However, credentials were found at:"
-        putStrLn $ "  " ++ fp
+        logWarn "WARNING: You've set save-hackage-creds to false"
```

Comment:
Any particular reason to include this change here? It seems unrelated, and also seems to be the cause of quite a large diff here.

Reply:
We wanted to make sure the user is notified when a `HACKAGE_KEY` is found and used instead of username/password, even if the latter are also already stored in env variables.

It felt lazy (in the negative sense) to do it with another `putStrLn` while we could do it with a `logInfo`. And since we could do *this* with a `logInfo`, we reckoned with just a little more effort we could resolve the TODO from the import section - and leave the module a tiny bit better than we found it.

Julian (@hasufell) was kind enough to contribute this particular set of changes (commit 135826b). If you feel it would help in any way, please let me know and I'll extract that commit into a separate PR.

Alternatively, I can add a mention of this change to the title of this PR to make it less surprising to be found here.

Reply:
> We wanted to make sure the user is notified when a `HACKAGE_KEY` is found and used instead of username/password, even if the latter are also already stored in env variables.
> 
> It felt lazy (in the negative sense) to do it with another `putStrLn` while we could do it with a `logInfo`. And since we could do *this* with a `logInfo`, we reckoned with just a little more effort we could resolve the TODO from the import section - and leave the module a tiny bit better than we found it.
> 
> Julian (@hasufell) was kind enough to contribute this particular set of changes (commit 135826b). If you feel it would help in any way, please let me know and I'll extract that commit into a separate PR.
> 
> Alternatively, I can add a mention of this change to the title of this PR to make it less surprising to be found here.

Basically, there's a TODO in the module already:

```
 
 ​import​           ​System.IO​                             (​putStrLn​, ​putStr​, ​print​) ​--​ TODO remove putStrLn, use logInfo
```

Reply:
It's not a question of whether it should be changed or not, but of mixing together two unrelated changes into a single PR. It makes it difficult to ensure only relevant code is changing and that no regressions are being introduced. For example, I noticed in the change here that moving to `logInfo` breaks the behavior of having a prompt on the same line without a following newline. It wasn't clear if this is intrinsic to the behavior of this feature, and the change in behavior wasn't called out at all.

Reply:
I see, thanks for pointing that out.